app/src/org/commcare/activities/GeoPointActivity.java

```diff
@@ -67,36 +89,48 @@ protected void onCreate(Bundle savedInstanceState) {
     @Override
     protected void onPause() {
         super.onPause();
-
-        // stops the GPS. Note that this will turn off the GPS if the screen goes to sleep.
-        if (ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED ||
-                ContextCompat.checkSelfPermission(this, Manifest.permission.ACCESS_COARSE_LOCATION) == PackageManager.PERMISSION_GRANTED) {
-            locationManager.removeUpdates(this);
-        }
-
-        // We're not using managed dialogs, so we have to dismiss the dialog to prevent it from
-        // leaking memory.
-        if (locationDialog != null && locationDialog.isShowing()) {
```

Comment:
shouldn't we still `dismiss` the dialog ?

Reply:
Aha! Thanks for catching this.